src/crush/CrushCompiler.cc

```diff
@@ -608,6 +608,10 @@ int CrushCompiler::parse_rule(iter_t const& i)
   }
 
   int ruleset = int_node(i->children[start]);
+  if (ruleset_name.count(ruleset)) {
+    err << "ruleset '" << ruleset << "' already defined in rule '" << ruleset_name[ruleset] <<"'\n"<< std::endl;
+    return -1;
```

Comment:
@tchaikov Hi! I think we want to do the same thing.But I  forbid the repeat  ruleset earlier.If the user have define the same ruleset I will forbid  him to generate crushmap from crushmap.txt.